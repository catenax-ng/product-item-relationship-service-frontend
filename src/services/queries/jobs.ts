import { useMutation } from "@tanstack/react-query";
import { useQuery } from "../../lib/react-query";
import { IRSRequestBody } from "../../types/jobs";
import { getCurrentEnvironment } from "../../utils/sessionStorageHandling";
import { selectAPI } from "../api/jobs.api";

export const useFetchJobById = ({ id }: { id: string }, refetchInterval: false | number = false) => {
  const serverEnv = getCurrentEnvironment();
  const jobAPI = selectAPI(serverEnv);
  return useQuery(["jobs", serverEnv, id], () => jobAPI.fetchJobById(id), {
    refetchInterval,
  });
};

export const useFetchJobs = (page: number, refetchInterval: false | number = false) => {
  const serverEnv = getCurrentEnvironment();
  const jobAPI = selectAPI(serverEnv);
  return useQuery(["jobs", serverEnv, page], () => jobAPI.fetchJobs(page), { refetchInterval });
};

export const useCancelJobs = () => {
  const serverEnv = getCurrentEnvironment();
  const jobAPI = selectAPI(serverEnv);
  return useMutation(jobAPI.cancelJob);
};

export const useCreateJob = () => {
  const serverEnv = getCurrentEnvironment();
  const jobAPI = selectAPI(serverEnv);
  return useMutation({
    mutationFn: (data: IRSRequestBody) => {
      return jobAPI.createJob(data);
    },
  });
};
