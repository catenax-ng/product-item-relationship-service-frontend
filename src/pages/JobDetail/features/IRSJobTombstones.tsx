import { Box, Divider, useTheme } from "@mui/material";
import dayjs from "dayjs";
import uniqueId from "lodash/uniqueId";
import { DetailGrid } from "../../../components/DetailGrid";
import { ErrorOutlineIcon, useTranslation } from "../../../lib";
import { Tombstone } from "../../../types/jobs";

export const IRSJobTombstones: React.FC<{ tombstones: Tombstone[] }> = ({ tombstones }) => {
  const { t } = useTranslation();
  const theme = useTheme();

  return (
    <section>
      <Box className="irs-tombstones-details">
        <Box className="irs-tombstones-details-header">
          <Box
            style={{
              display: "inline-block",
              color: theme.palette.error.light,
              marginTop: 20,
            }}
          >
            <ErrorOutlineIcon
              style={{
                fontSize: 50,
                float: "left",
                verticalAlign: "middle",
                marginTop: 10,
              }}
            />
            <h2 style={{ float: "left", marginLeft: 10 }}>{t("content.irs.dialog.submodelTombstones.title")}</h2>
          </Box>
        </Box>
        <Box className="irs-tombstones-details-content">
          {tombstones.map((stone) => {
            return (
              <Box key={`${uniqueId(stone.catenaXId)}`}>
                <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
                <DetailGrid
                  topic={t("content.irs.dialog.submodelTombstones.lastAttempt") + ":"}
                  content={dayjs(stone.processingError.lastAttempt).format("YYYY-MM-DD HH:mm:ss")}
                />
                <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
                <DetailGrid
                  topic={t("content.irs.dialog.submodelTombstones.errorDetail") + ":"}
                  content={stone.processingError.errorDetail}
                />
              </Box>
            );
          })}
        </Box>
      </Box>
    </section>
  );
};
