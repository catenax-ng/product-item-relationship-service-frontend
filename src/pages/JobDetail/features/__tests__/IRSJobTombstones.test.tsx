import { JobsDemoDataSuccess } from "../../../../services/api/__test__/jobs.exampleData";
import { render } from "../../../../testing/test-utils";
import { IRSJobTombstones } from "../IRSJobTombstones";

it("renders the tombstone information", () => {
  const tombstones = JobsDemoDataSuccess.tombstones;
  const { container } = render(<IRSJobTombstones tombstones={tombstones} />);
  expect(container).toMatchSnapshot();
});
