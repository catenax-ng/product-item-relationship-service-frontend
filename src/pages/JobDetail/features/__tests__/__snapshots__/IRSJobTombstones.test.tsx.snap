// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the tombstone information 1`] = `
<div>
  <section>
    <div
      class="irs-tombstones-details MuiBox-root css-0"
    >
      <div
        class="irs-tombstones-details-header MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-0"
          style="display: inline-block; color: rgb(239, 83, 80); margin-top: 20px;"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ErrorOutlineIcon"
            focusable="false"
            style="font-size: 50px; float: left; vertical-align: middle; margin-top: 10px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <h2
            style="float: left; margin-left: 10px;"
          >
            content.irs.dialog.submodelTombstones.title
          </h2>
        </div>
      </div>
      <div
        class="irs-tombstones-details-content MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-0"
        >
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.dialog.submodelTombstones.lastAttempt:
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              2022-02-03 14:48:54
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.dialog.submodelTombstones.errorDetail:
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              Details to reason of Failure
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
